### DISCUSSION TITLE
PlatformIO Debugging Peripherals are loading forever

### DISCUSSION THREAD

**Discussion by Tobias3006** [earlephilhower/arduino-pico#1579] [Q&A]
Hi, I'm trying to debug an Pi Pico with the official Debug Probe (both SWD and Serial-Interface is wired) using VSC with PlatformIO extension. Everything works fine but the "Peripherals" tab is loading in an endless loop. Can anyone help me with this issue? 

For Testing I'm using only a basic Blink-Program: 
```c++
# include <Arduino.h>

int i = 0;

void setup() {
    pinMode(25, OUTPUT);
}

void loop() {
    
    digitalWrite(25, HIGH);
    delay(500);
    digitalWrite(25, LOW);
    delay(500);
    i++;
}
``` 

And this is my platform.ini configuration: 

```
[env:pico]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = pico
framework = arduino
board_build.core = earlephilhower
debug_tool = cmsis-dap
upload_protocol = cmsis-dap
build_type = debug
debug_build_flags = -O0 -ggdb3 -g3
``` 

Image of Error:

![image](https://github.com/earlephilhower/arduino-pico/assets/13836438/0aa93f52-9f1e-45ad-b8a9-105067633168)

**Comment by maxgerhardt** [comment]
That shouldn't happen. Are you using the lates version of the Rapberry pi platform, the PlatformIO core, the PlatformIO VSCode extension and VSCode? To take care of the first two, you can [CLI](https://docs.platformio.org/en/latest/integration/ide/vscode.html#platformio-core-cli) -> 
```
pio upgrade --dev
pio pkg update -g -p raspberrypi
```

**Reply by Tobias3006** [reply]
Yes all components are up to date: 

- Raspberry Pi platform: development version
- PlaformIO core: 6.1.9rc1
- PlatformIO VSCode extension: 3.4.4
- VSCode: 1.79.2

**Reply by maxgerhardt** [reply]
I'll double check this at home.
It might actually be that this **never** worked because PlaformIO's SVD parser does have problems (https://github.com/platformio/platformio-vscode-ide/issues/2368, https://github.com/platformio/platformio-vscode-ide/issues/2336, https://github.com/platformio/platformio-vscode-ide/issues/1675)

**Comment by LinusHeu** [comment]
I can confirm, in the bottom right it shows the following error message:
"Unable to parse SVD file: TypeError: Cannot read properties of undefined (reading '0')"'

As far as I remember, it never worked.

**Reply by maxgerhardt** [reply]
Thanks for confirming, I've pushed the issue https://github.com/platformio/platformio-vscode-ide/issues/1675 further.
